kotlintest-assertions/src/main/kotlin/io/kotlintest/assertions/sysprops.kt

```diff
@@ -0,0 +1,34 @@
+package io.kotlintest.assertions
+
+import java.util.*
+
+fun <T> withSystemProperty(key: String, value: String, thunk: () -> T): T {
```

Comment:
Perhaps allow value to be nullable? One might want to override a specific System property to null